lms/djangoapps/courseware/views.py

```diff
@@ -726,6 +726,23 @@ def registered_for_course(course, user):
         return False
 
 
+def get_cosmetic_display_price(course, registration_price):
+    """
+    Return Course Price as a string preceded by correct currency, or 'Free'
+    """
+    currency_symbol = settings.PAID_COURSE_REGISTRATION_CURRENCY[1]
+
+    price = course.cosmetic_display_price
+    if registration_price > 0:
+        price = registration_price
```

Comment:
So, if a registration price is passed in, it overrides the cosmetic display price? That seems strange to me -- I would expect that if a course has a cosmetic display price defined, then it would override everything else.